recipe-app/src/RecipeInfo.js

```diff
@@ -0,0 +1,97 @@
+import React, {useState, useEffect} from "react";
+import styled from "styled-components";
+import RecipeForm from "./RecipeForm";
+import {NavLink, Redirect} from "react-router-dom";
+import useRecipeState from "./hooks/useRecipeState";
+
+const StyledRecipe = styled.div`
+    border: black 5px double;
+    padding: 5px;
+    margin: 5px;
+    background: lightgray;
+    text-align: left;
+`
+
+const UnorderedList = styled.ul`
+    font-size: 10pt;
+    display: block;
+    text-align: left;
+`
+
+const Button = styled.button`
+    float: right;
+`
+
+function buildIngredients(ingredients) {
+    return (
+        <div>
+            <p>Ingredients:</p>
+            <UnorderedList>
+                {ingredients.map((ingredient, idx) => (
+                    <li key={idx}>{ingredient}</li>))}
+            </UnorderedList>
+        </div>
+    );
+}
+
+function RecipeInfo(props) {
+    let recipeId = props.match.params.id;
+
+    let [loading, setLoading] = useState(true);
+
+    let {recipe, loadRecipe, updateRecipe} = useRecipeState(
```

Comment:
`loading` could be handled internally to `useRecipeState` and returned here, so that you wouldn't need to pass `setLoading` as parameter